src/Sulu/Bundle/AdminBundle/Resources/js/containers/Sidebar/stores/SidebarStore.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import type {SidebarConfig, Sizes} from '../types';
+import {DEFAULT_SIZE, SIZES} from '../types';
+
+class SidebarStore {
+    @observable view: ?string;
+    @observable props: Object;
+    sizes: Array<string>;
+
+    @observable size: ?string;
+
+    constructor() {
+        this.clearConfig();
+    }
+
+    @action setConfig(config: SidebarConfig) {
+        this.view = config.view;
+        this.props = config.props || {};
+        this.sizes = config.sizes || SIZES;
+
+        let defaultSize = config.defaultSize || DEFAULT_SIZE;
+        if (!this.size || -1 === this.sizes.indexOf(this.size)) {
+            this.size = defaultSize;
```

Comment:
I would use `setSize` here, so that you also get an error message when the `DEFAULT_SIZE` does not match any of the sizes.